packages/javascript/bh-shared-ui/src/views/UserProfile/UserProfile.tsx

```diff
@@ -46,8 +46,19 @@ const UserProfile = () => {
     );
 
     const updateUserPasswordMutation = useMutation(
-        ({ userId, secret, needsPasswordReset }: { userId: string; secret: string; needsPasswordReset: boolean }) =>
+        ({
+            userId,
+            secret,
+            needsPasswordReset,
+            currentSecret,
+        }: {
+            userId: string;
+            currentSecret?: string;
+            secret: string;
+            needsPasswordReset: boolean;
+        }) =>
             apiClient.putUserAuthSecret(userId, {
+                ...(currentSecret && { current_secret: currentSecret }),
```

Comment:
this line can be simplified to:
`current_secret: currentSecret`.  

you can safely remove the && operator and the ... operator.  on mutations where currentSecret is undefined, the field is omitted from the post request body.